### PR TITLE
feat(components): add data-table

### LINKED ISSUE #141: component: data-grid/table
a new component that can display data as a table and should include:

- display JSON as table
- sorting
- filtering
- pagination or endless scrolling
- custom cell renderer (optional for MVP)
- server-side-rendering

### PR DESCRIPTION
**Pull request by devCrossNet**
closes #141

### What is accomplished by your PR?
This PR adds a new DataTable component which includes following features:

- define column schema (not-displaying certain columns, rename titles, rearrange appearance)
- display JSON as table
- sorting of rows, one column at a time
- filtering
- pagination
- server-side-rendering

screen shot 2018-06-10 at 12 20 11

### Is there something controversial in your PR?
nope

### Link to the Issue
#141 

# Checklist

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing
- [x] Add new passing unit tests to cover the code introduced by your PR

### CODE REVIEW

File: src/app/shared/components/VueDataTable/VueDataTable.vue [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+      count() {
+        return this.filteredData.length;
+      },
+    },
+    data(): any {
+      return {
+        sortKey:       null,
+        sortDirection: 'asc',
+        currentPage:   0,
+        maxPages:      1,
```
Comment (devCrossNet) [inline src/app/shared/components/VueDataTable/VueDataTable.vue]: computed prop, to remove the watcher and cache the result

File: src/app/shared/components/VueDataTable/VueDataTable.vue [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+      },
+    },
+    computed:   {
+      filteredData() {
+        if (this.searchTerm.length === 0) {
+          return this.data;
+        }
+
+        const searchRegex: RegExp = new RegExp(`${this.searchTerm}`, 'gmi');
+        const filter = (columns: any[]) => {
```
Comment (devCrossNet) [inline src/app/shared/components/VueDataTable/VueDataTable.vue]: object, not array

File: src/app/shared/components/VueDataTable/VueDataTable.vue [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+          header.sortKey = key;
+
+          return header;
+        });
+      },
+      columnWidth() {
+        return `${100 / this.columns.length}%`;
+      },
+      rows() {
+        return this.displayData.map((row: any[]) => {
```
Comment (devCrossNet) [inline src/app/shared/components/VueDataTable/VueDataTable.vue]: object, not array

File: src/app/shared/components/VueDataTable/VueDataTable.stories.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+story.add('Default', () => ({
+  i18n,
+  components: { VueDataTable },
+  data() {
+    return {
+      header: dataTableHeaderFixture,
+      data:   dataTableDataFixture,
+    };
+  },
+  template:   `<vueDataTable :header="header" :data="data" placeholder="Search" @click="action" />`,
```
Comment (40818419) [inline src/app/shared/components/VueDataTable/VueDataTable.stories.ts]: vue-data-table

File: src/app/shared/components/VueDataTable/VueDataTable.stories.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+story.add('All Props', () => ({
+  i18n,
+  components: { VueDataTable },
+  data() {
+    return {
+      header: dataTableHeaderFixture,
+      data:   dataTableDataFixture,
+    };
+  },
+  template:   `<vueDataTable :header="header" :data="data" :show-search="false" :page="1" :max-rows="10" @click="action" />`,
```
Comment (40818419) [inline src/app/shared/components/VueDataTable/VueDataTable.stories.ts]: vue-data-table

**Pull request merged**